core/config/config.go

```diff
@@ -146,6 +146,12 @@ type Config struct {
 	// If value is not set, as a fallback ChainConfigs keys are used.
 	XChainBlockchains []uint64
 
+	// EtherBasedXChainBlockchains is a list of chain IDs that are allowed to be used in xChain checks.
+	// If value is not set, the chains in GetDefaultBlockchainInfo() that use ether as a native fee currency
+	// will be used. This list will always be a subset of XChainBlockchains - exteraneous chain IDs will be discarded.
+	// TODO: this value is going to be moved on-chain so same setting is shared between all nodes and clients.
+	EtherBasedXChainBlockchains []uint64
```

Comment:
This PR looks good, however it is not clear why this config param is required: would it make sense just to filter `XChainBlockchains` by `IsEtherBased` and get this list this way? I.e. do we have a scenario when chain is Ehter-based, but we don't want it in the balance check?

Reply:
If GetDefaultBlockchainInfo captures information we plan to move on-chain, then I think that's fine, I can refactor.